tests/client/auth/test_auth_token_refresh_chain.py

```
@@ -0,0 +1,443 @@
+    # 5. 401 mid-flight reactive refresh
+    # ------------------------------------------------------------------
+    @pytest.mark.asyncio
+    async def test_401_mid_flight_reactive_refresh(self) -> None:
+        """ensure_authenticated passes, but API returns 401 — decorator recovers."""
+
+        class MidFlightService:
+            def __init__(self) -> None:
+                self.call_count = 0
+                self.refresh_called = False
+                self.clear_auth_token_called = False
+                self.auth_token = MagicMock(refresh_token="valid")
```

Ruff (0.15.9)

[error] Possible hardcoded password assigned to argument: "refresh_token"

(S106)